src/dataset-builder/DatasetBuilder.ts

```diff
@@ -653,6 +654,7 @@ export const DatasetBuilderContents = ({
         conceptSets: allConceptSets,
         valuesSets: selectedValues,
         onDismiss: () => setRequestingAccess(false),
+        datasetId: dataset.id,
```

Comment:
Looking at other uses of `dataset` here I see we sometimes use `dataset?.` to get at dataset fields. But I think the `?.` code is incorrect since `dataset` is always defined. If you agree and want to, you could fix those cases now too.